controller/controller.mustache

```diff
@@ -64,13 +65,22 @@ module.exports = {
   create: ({{name}}) => {
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
+    {{name}}.rev = uuid.v4();
     return r.table('{{pluralName}}').insert({{name}}, {returnChanges: true}).run()
     .then(rethinkdb.firstChange);
   },
   update: ({{name}}) => {
+    if (!{{name}}.rev) return Promise.reject(new httpError('BAD_REQUEST', 'No rev provided'));
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
-    return r.table('{{pluralName}}').update({{name}}, {returnChanges: true}).run()
+    return r.table('{{pluralName}}')
+    .get({{name}}.id)
+    .update(
+      rethinkdb.ifRevMatches({{name}}),
+      {returnChanges: true}
+    )
+    .run()
+    .then(rethinkdb.checkRevError)
```

Comment:
Oh yeah, maybe not that one.  Does rethink support chaining a `.catch`?  That might be more appropriate than `.then`, but I'm not sure if that's available.

Where I'm getting to with the above `conflict` function suggestion (disregarding my comment of amalgamating both the added rethink functions, as I just mentioned), is that it seems more appropriate to try and use that for the intended functionality.